Objective-C/Tests/ReplicatorTest.m

```diff
@@ -2119,6 +2119,141 @@ - (void) testPullDeletedDocWithFilter: (BOOL)isContinuous {
     AssertEqual(otherDB.count, 0u);
 }
 
+
+#pragma mark stop and restart the replication with filter
+
+
+- (void) testStopAndRestartPushReplicationWithFilter {
+    // Create documents
+    NSError* error;
+    CBLMutableDocument* doc1 = [[CBLMutableDocument alloc] initWithID: @"doc1"];
+    [doc1 setString: @"pass" forKey: @"name"];
+    Assert([self.db saveDocument: doc1 error: &error]);
+    
+    NSMutableSet<NSString*>* docIds = [NSMutableSet set];
+    id target = [[CBLDatabaseEndpoint alloc] initWithDatabase: otherDB];
+    CBLReplicatorConfiguration* config = [self configWithTarget: target
+                                                           type: kCBLReplicatorTypePush
+                                                     continuous: YES];
+    config.pushFilter = ^BOOL(CBLDocument* document, CBLDocumentFlags flags) {
+        AssertNotNil(document.id);
+        [docIds addObject: document.id];
+        
+        // allow all docs with `name = pass`
+        return [[document stringForKey: @"name"] isEqualToString: @"pass"];
+    };
+    
+    repl = [[CBLReplicator alloc] initWithConfig: config];
+    [repl start];
+    [self waitForReplicatorListenerToStop];
```

Comment:
You can use `[self runWithReplicator: replicator errorCode: 0 errorDomain: nil];` so that `waitForReplicatorListenerToStop` will not needed.